services/api/src/routes/users.js

```diff
@@ -52,6 +54,20 @@ router
       };
     }
   )
+  .get(
+    '/:userId/token',
+    requirePermissions({ endpoint: 'users', permission: 'write', scope: 'global' }),
+    async (ctx) => {
+      const { user } = ctx.state;
+      if (!user.authTokenId) {
```

Comment:
Not loving this ... 

Alternative i could create a new type of token, that we validate differently.

Reply:
First, is there actually a need to attach the token itself to the user being impersonated as opposed to the user impersonating with, for example, `impersonatedTokenId` or something?

Also I'm sorry to make this more complicated but I'm wondering if this isn't a good opportunity to also look at multi-device logins as I have a feeling that it dovetails with this.

To just run this thought experiment: what if `authTokenId` became `authTokenIds` and we simply kept them as an array? Setting aside more complex questions of expiration this would allow in theory users to be logged on different devices. It ALSO (I'm thinking) would allow the impersonating user to have their own login token as well as tokens of those other users that they are impersonating. I'm not suggesting that we allow them to impersonate more than one user at a time, but as an approach it seems like this might be more forward thinking possibly?

Reply:
I think you might be right re authTokenIds, so perhaps i will do that first, that would make this a bit more streamline. Will create a separate branch for that. 

I think it would make sense that if the admin user logs out we would invalidate all the tokens, ever created by the admin, regardless of token type (impersonated / regular auth). 

On top of that I think will add some auditLogs to this endpoint + further support in the auditLogs (might also be a seperate PR) 

`I'm not suggesting that we allow them to impersonate more than one user at a time, but as an approach it seems like this might be more forward thinking possibly?`
Thats totally fine to allow. But we should consider if we can prevent more than X tokens created in a given timeframe.
And perhaps put a max length on the number of authTokenIds you can have.

Reply:
yeah I think there's various constraints you might want to put on this ... probably it depends on the project too so I wouldn't mind leaving it open to start

Reply:
one other thing to think about... if this is meant to be only temporary (while the window is open) it would be kind of annoying to have a lot of impersonation tokens build up in this array... not sure what the best way to handle that would be

Reply:
should be handled now, the tokens will just expire now and removed from the user.authInfo